src/answers-headless.ts

```diff
@@ -226,30 +226,37 @@ export default class AnswersHeadless {
       context,
       referrerPageUrl
     };
-    const results = await this.core.verticalSearch(request);
+    const response = await this.core.verticalSearch(request);
     const latestResponseId = this.httpManager.getLatestResponseId('verticalQuery');
     if (thisRequestId < latestResponseId) {
-      return results;
+      return response;
     }
     this.httpManager.setResponseId('verticalQuery', thisRequestId);
-    this.stateManager.dispatchEvent('vertical/setResults', results);
-    this.stateManager.dispatchEvent('query/setQueryId', results.queryId);
+    this.stateManager.dispatchEvent('query/setQueryId', response.queryId);
     this.stateManager.dispatchEvent('query/setLatest', query);
-    this.stateManager.dispatchEvent('filters/setFacets', results.facets);
-    this.stateManager.dispatchEvent('spellCheck/setResult', results.spellCheck);
-    this.stateManager.dispatchEvent('vertical/setAlternativeVerticals', results.alternativeVerticals);
-    this.stateManager.dispatchEvent('location/setLocationBias', results.locationBias);
-    this.stateManager.dispatchEvent('query/setSearchIntents', results.searchIntents || []);
-    this.stateManager.dispatchEvent('location/setLocationBias', results.locationBias);
+    this.stateManager.dispatchEvent('filters/setFacets', response.facets);
+    this.stateManager.dispatchEvent('spellCheck/setResult', response.spellCheck);
+    this.stateManager.dispatchEvent('location/setLocationBias', response.locationBias);
+    this.stateManager.dispatchEvent('query/setSearchIntents', response.searchIntents || []);
+    this.stateManager.dispatchEvent('location/setLocationBias', response.locationBias);
+    this.stateManager.dispatchEvent('directAnswer/setResult', response.directAnswer);
     this.stateManager.dispatchEvent('vertical/setSearchLoading', false);
-    this.stateManager.dispatchEvent('directAnswer/setResult', results.directAnswer);
-    return results;
+    this.stateManager.dispatchEvent('vertical/setAllResultsForVertical', response.allResultsForVertical);
```

Comment:
what do you think about trying the single reducer you mentioned, for spreading the response into vertical state? feels like it might be a little nicer that way, also maybe slightly more performant since each dispatch loops through all the listeners (this is probably negligible though)

Reply:
Sure, I can give it a shot